### ISSUE TITLE
[Request] Upgrade default API version of Azure OpenAI

### ISSUE THREAD

**Issue by kmahyyg** [lobehub/lobe-chat#991]
### 🥰 需求描述

Microsoft had sent email that informed Azure OpenAI users to move to "2023-12-01-preview", all legacy versions will be deprecated on Apr 2024.

I'd like to confirm if this application support this version of Azure OpenAI API call or not? So that I could continuously use this application without break of the service.

Thanks for your excellent work.

**Note: MSFT also mentioned that `gpt-4-vision-preview` MUST use this version of API on its document https://learn.microsoft.com/en-us/azure/ai-services/openai/reference . **

### 🧐 解决方案

Upgrade default Azure OpenAI API version from "2023-08-01-preview" to "2023-12-01-preview".

### 📝 补充信息

Microsoft Email Screenshot:
![2024-01-09 17:30:57](https://github.com/lobehub/lobe-chat/assets/16604643/b8401cf0-d067-4ec7-94e5-504d7e4b9875)

**Comment by arvinxx** [comment]
PR welcome~ https://github.com/lobehub/lobe-chat/blob/05209a23095d6ece06b51d53b718118149990651/src/app/api/openai/createBizOpenAI/createAzureOpenai.ts#L19

**Comment by kmahyyg** [comment]
After carefully review the swagger spec between two versions:
- https://github.com/Azure/azure-rest-api-specs/raw/main/specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-08-01-preview/generated.json
and
- https://github.com/Azure/azure-rest-api-specs/raw/main/specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-12-01-preview/generated.json

I determined that for Azure OAI part, only DALL-E and audio transcription API has been fully updated. 

However, it seems that function-call feature has been changed a bit. Since I'm not a frontend developer, after personal assessment, I'm afraid that I cannot open a PR now.

Anyway, still thanks for your response and excellent work.
